src/coreclr/interpreter/compiler.cpp

```diff
@@ -2163,10 +2163,14 @@ int32_t InterpCompiler::GetDataItemIndexForHelperFtn(CorInfoHelpFunc ftn)
     CORINFO_CONST_LOOKUP ftnLookup;
     m_compHnd->getHelperFtn(ftn, &ftnLookup);
     void* addr = ftnLookup.addr;
-    if (ftnLookup.accessType == IAT_PVALUE)
+    if (ftnLookup.accessType == IAT_VALUE)
     {
-        addr = (void*)((size_t)addr | INTERP_INDIRECT_HELPER_TAG);
+        // We can't use the 1 bit to mark indirect addresses because it is used for real code on arm32 (the thumb bit)
+        // So instead, we mark direct addresses with a 1 and then on the other end we will clear the 1 and re-set it as needed for thumb
```

Comment:
On win-x86, there is no guaranteed alignment for function entrypoints. Tagging schemes like this do not work on win-x86 for native code addresses.

Reply:
I will remove all uses of tagged helpers in a follow-up PR. It doesn't sound like they are portable.